### PR TITLE
Fix loadPackage on relative urls

### LINKED ISSUE #3011: Relative URLs in pyodide.loadPackage
## 🐛 Bug

The documentation states that [pyodide.loadPackage](https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.loadPackage) supports relative URLs. I'm trying to load an out-of-tree wheel from my local webserver, but this doesn't seem to work out well.

### To Reproduce

```js
await pyodide.loadPackage("dist/igraph-0.9.11-cp310-cp310-emscripten_3_1_14_wasm32.whl");
```
or
```js
await pyodide.loadPackage("./dist/igraph-0.9.11-cp310-cp310-emscripten_3_1_14_wasm32.whl");
```
Pyodide tries to load the wheel from `https://cdn.jsdelivr.net/pyodide/v0.21.1/full/dist/igraph-0.9.11-cp310-cp310-emscripten_3_1_14_wasm32.whl`.

### Expected behavior

Load the wheel from the relative URL.

### Environment

- Pyodide Version: 0.21.1
- Browser version: Firefox ESR 91.12.0, Chromium 104.0.5112.101
- Any other relevant information:

### PR DESCRIPTION
**Pull request by hoodmane**
### Description

This fixes #3011.

We attempted to make a test for this, but in all of our tests `document.location === indexURL`. Our logic incorrectly used `indexURL` instead of `document.location`.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

**Comment by hoodmane** [comment]
@ryanking13 any clue what's going wrong with the node tests here?

**Comment by hoodmane** [comment]
Nevermind, error reproduces locally.

File: src/tests/test_package_loading.py [hunk 1]
```diff
@@ -58,10 +61,35 @@ def test_load_from_url(selenium_standalone, web_server_secondary, active_server)
     selenium.run("import pytz")
 
 
-def test_load_relative_url(selenium_standalone):
-    print(get_pytz_wheel_name())
-    selenium_standalone.load_package(f"./{get_pytz_wheel_name()}")
-    selenium_standalone.run("import pytz")
+def test_load_relative_url(request, runtime, web_server_main, playwright_browsers):
+    from tempfile import TemporaryDirectory
```
Comment (ryanking13) [inline src/tests/test_package_loading.py:65]: Maybe use `tmp_path` fixture instead?

File: src/tests/test_package_loading.py [hunk 2]
```diff
@@ -58,10 +61,35 @@ def test_load_from_url(selenium_standalone, web_server_secondary, active_server)
...
+
+    with TemporaryDirectory() as temp, spawn_web_server(temp) as web_server:
+        directory = Path(temp)
+        url, port, _ = web_server_main
+        test_html = (ROOT_PATH / "src/templates/test.html").read_text()
```
Comment (ryanking13) [inline src/tests/test_package_loading.py:70]: `DIST_PATH / "test.html"` would work, as we vendors it.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane!

File: src/tests/test_package_loading.py (hunk 1 above)
Comment (hoodmane) [reply src/tests/test_package_loading.py:65]: I always forget about this fixture.

**Review by rth — APPROVED** [review]
Thanks for fixing it!

**Pull request merged**